### ISSUE TITLE
RangeError: /node_modules/gatsby-plugin-mdx/index.js: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by t2ca** [gatsbyjs/gatsby#34823]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Upgrading `@babel/core` to the latest version `7.17.3` will give you the following error

```
ModuleBuildError: Module build failed (from ./node_modules/gatsby/dist/utils/babel-loader.js):
  RangeError: /node_modules/gatsby-plugin-mdx/index.js: Maximum call stack size exceeded
```
This is an issue getting addressed here https://github.com/babel/babel/issues/14273

If you are currently having this issue with your gatsby project, there is a temporary solution by adding the following to your `package.json` file

```
  "resolutions": {
    "@babel/core": "7.17.2"
  }
```

### Reproduction Link

https://github.com/gatsbyjs/gatsby/

### Steps to Reproduce

1. yarn upgrade
2. gatsby develop

### Expected Result

no error

### Actual Result

```
ModuleBuildError: Module build failed (from ./node_modules/gatsby/dist/utils/babel-loader.js):
  RangeError: /node_modules/gatsby-plugin-mdx/index.js: Maximum call stack size exceeded
```

### Environment

```shell
System:
    OS: macOS 12.1
    Shell: 5.8 - /usr/local/bin/zsh
  Binaries:
    Node: 17.4.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.4.0 - /usr/local/bin/npm
```

**Comment by nicolo-ribaudo** [comment]
We released the fix in `@babel/core` 7.17.4, sorry!

**Comment by t2ca** [comment]
Thank you!
